code/datums/components/jam_receiver.dm

```diff
@@ -35,11 +35,37 @@
 	if (move_relay.depth == 0)
 		move_relay.RemoveComponent()
 		associated_relay = null
+	remove_self_from_list(associated_z)
 
 /datum/component/jam_receiver/proc/on_move(...)
 	SIGNAL_HANDLER
+	check_z_list_sanity()
 	check_jammed()
 
+/// gets z of the parent, but possibly not as 0
+/datum/component/jam_receiver/proc/get_sanitized_z()
+	var/atom/parent_atom = parent
+	while(parent_atom.z == 0 && !isarea(parent_atom.loc)) // we try to find a true Z value
+		parent_atom = parent_atom.loc
+	return parent_atom.z
```

Comment:
This doesn't need to be a proc, just get_turf(parent) and then use the z-value of that

Reply:
It's better to be a proc anyway because you can get null turf

Reply:
actually, you're right